pkg/transformers/refvars.go

```diff
@@ -41,49 +43,84 @@ func NewRefVarTransformer(
 	}
 }
 
+// replacePrimitiveType checks if the field is a string. If not it will return
+// the field. In case it is a string, it will expand the field, and perform
+// a deepCopy of the expanded value in case it is not of primitiv type.
+func (rv *RefVarTransformer) replacePrimitiveType(a interface{}) interface{} {
+	s, ok := a.(string)
+	if !ok {
+		// This field is not of string type.
+		// It can not contain a $(VAR)
+		return a
+	}
+
+	// This field can potientially contain a $(VAR)
+	expandedValue := expansion.Expand(s, rv.mappingFunc)
+
+	// Let's perform a deep copy if we didn't inline
+	// a primitive type
+	return deepCopy(expandedValue)
+}
+
+// replaceParentInline allows to inline the complex tree of a variable
+// at the same time it allows to replace and patch individual member of
+// that inlined tree.
+func (rv *RefVarTransformer) replaceParentInline(inMap map[string]interface{}) (interface{}, error) {
+	s, _ := inMap[parentInline].(string)
+
+	inlineValue := expansion.Expand(s, rv.mappingFunc)
+	newMap, ok := inlineValue.(map[string]interface{})
+	if !ok {
+		log.Printf("inlining issue with %s", inlineValue)
```

Comment:
Traditionally, kustomize doesn't log errors and continue, opting to instead just fail the build with an error.

To the end use, logged errors and still emitting output just raises questions like "is this right?"

Reply:
changed to be aligned with kustomize traditional behavior. Removed the log statement and returned an error